### DISCUSSION TITLE
Nzyme-Tap install on OpenWrt? Need help, Please!

### DISCUSSION THREAD

**Discussion by purevida2001** [nzymedefense/nzyme#993] [Questions]
Has anyone figured out how to install nzyme-tap on OpenWrt? I am trying to install the Debian package, but can't install dpkg on OpenWrt. Is there a solution or does anyone know how to convert the .deb file into a format that OpenWrt will allow to install? I am not good at code writing but would really like this to work with my existing setup.

**Comment by lennartkoopmann** [comment]
Hi,

can you describe what doesn't work? What hardware are you using and what CPU architecture is it?

Thanks!

**Comment by purevida2001** [comment]
I have the latest Nzyme Node software running on a server with ubuntu 22.04 workstation. I am trying to reuse some airtight C-75 POE access points because I already have a handful of them. I found an OpenWrt image that will install on these access points. I am willing to use a different access point but will need to make sure they are POE so I dont have to run power to each device running Nzyme-tap. I was able to run a git request and pull the nzyme-tap.deb image from the site, but it seems that the OpenWrt access point will not allow me to install the dpkg software to install nzyme-tap.deb files. There is limited memory and storage on these access points (details below) I hope you can help. 

System info
Hostname	OpenWrt
Model	AirTight Networks C-75
Architecture	Qualcomm Atheros QCA9558 ver 1 rev 0
Target Platform	ath79/generic
Firmware Version	OpenWrt 23.05.2 r23630-842932a63d / LuCI openwrt-23.05 branch git-23.306.39416-c86c256
Kernel Version	5.15.137
Local Time	2024-01-15 14:48:10
Uptime	3d 23h 15m 57s
Load Average	1.28, 1.35, 1.36
Memory
Total Available	46.00 MIB / 119.20 MIB (38%)
Used	65.99 MIB/ 119.20 MIB (55%)
Cached	32.09 MIB /119.20 MIB (26%)
Storage
Disk space	10.68 MIB / 25.44 MIB (41%)

SomeUser@OpenWrt:~# uname -a
Linux OpenWrt 5.15.137 #0 Tue Nov 12 13:38:11 2023 mips GNU/Linux

**Reply by lennartkoopmann** [reply]
There are no prepackaged nzyme builds for `ath79` and there are currently no plans to support it. Also, I don't believe that OpenWrt lets you install `.deb` packages. 

For this to work, you'd have to compile the tap to `ath79` yourself and take care of the packaging as well. I would not recommend to try to do that, because 120 MB memory total, with only 46 MB free/available will be no fun at all for something as demanding as the nzyme tap. It may just work, but be a constant struggle.

I would recommend using a Raspberry Pi 3 with the POE hat. The Raspberry Pi 3 is cheap and will be significantly faster than your current device.

**Reply by purevida2001** [reply]
Lennart, thank you for the fast replies. I will have to look into getting a few POE hats, I have a few Raspberry PI 4 devices. Do you know of any wifi access point that currently supports Nzyme-Tap install (Netgear, TP-Link, GLi-Net)? I only ask because I think if I place a bunch of Raspberry PI devices around the building, they will start to disappear in people's pockets.

**Reply by lennartkoopmann** [reply]
I believe that the larger GLinet devices would be large enough, but you'd still have to compile your own taps to support architecture and packaging. We have also seen issues with drivers that are very liberally passing bad frames, leading to all sorts of issues in the internal routing and processing.

We are working on official Raspberry Pi kits with cases and mounts. The issue of device theft is very interesting. In this case, the mounts are designed to be stealth and not easily removed. This is a few weeks out. You could start by provisioning Raspberry Pis now and improve the deployment with our kits later.
